### ISSUE TITLE
Feature Request: Financial Difficulty Setting

### ISSUE THREAD

**Issue by desFritz** [endless-sky/endless-sky#7079]
### Problem Description

I almost feel bad to submit the third feature request, but I am willing to submit content in the future, too.  ;-)
It's just every time I think "Hey, that would be a nice thing..." I run into a missing function/feature etc. to implement it.
And since I can't code C**

Short: 
Is there not a way to change the payment amount via a global setting/slider?

Long:
Making money is easy. You can mine, you can haul cargo and become rich rather fast. Maybe others would prefer more of a challenge, clower pace, or have it even easier?

From my understanding I could now edit all "job..txt" files and change the payment to half etc. or double it. For every job...
Sure, python/sed/awk & co. could do it, but maybe others would find such a feature useful too?

### Related Issue Links

[x] searched the issues back to 2016 and on steam, but searching for difficulty lists people complaining about game difficulty etc. 

### Desired Solution

Like 'some' games it might be wise to add difficulty settings to the menu.

![grafik](https://user-images.githubusercontent.com/109269346/182162242-0f36c6bb-539d-47f3-9992-51c82fa9da1f.png)

Whenever the game accesses "payment" from jobs/missions it will multiply it with the modifier from the settings menu before printing it to screen, oder modifying the credits value from the bank.

Slide to 2.0 in settings and payment 10,000 will turn into 20,000.
Slide to 0.5 in settings and payment 10,000 will turn into 5,000.

Most likely it needs to go to the savegame file, so it gets saved per captain/profile.

"Screenshot" should most likely say "job/mission payment", not when you sell things in shipyard/outfitter etc.
The payment from mining could be finetuned to your own likings as well.

### Alternative Approaches

None from my side.

**Comment by Zitchas** [comment]
That's actually a pretty good idea. Have a single multiplier that can be set by the player, recorded in their save file instead of preferences; and serve as multiplier anytime the player is given credits through missions. That'd be a good way to let people adjust the speed of the income growth.

This is also something that may fit better as an element of the gamerules system discussed by Amazinite here:
https://github.com/endless-sky/endless-sky/issues/5556

**Comment by quyykk** [comment]
Increasing difficulty by increasing the grind is lazy and not good game design. As such, I 'd prefer it if difficulty was done by doing doing stuff like:

- better/worse AI
- make the AI always/never disable ships (i.e. in hard mode they never leave hostile disabled ships alone and vice versa).
- better/worse loadouts for ships
- easier/harder campaign battles
- etc.

**Comment by Zitchas** [comment]
For overall game difficulty, I agree, Quyyk. Where this idea has merit is letting players specifically calibrate their grind. Some people like the space-trucker lifestyle, others feel the need to just snowball it to get a few billion and then dispense with worrying about money ever again.

This isn't really about making the game harder or easier. I personally don't find that my income really has much impact on the difficulty, especially if I'm flying a solo ship. I find that I end up with an "optimized the best I can manage" ship fairly quickly, and it doesn't seem to have a huge impact on the difficulty of the combat.

I think it would be more appropriate to label this as "A setting to change the game's financial pacing." It's not difficulty, it's pace. Do player's need to run hundreds of jobs to be able to afford to upgrade to the next ship up? Or just a handful?

**Comment by leklachu** [comment]
A problem I see with that is if the game forces you to grind, to a certain point, it feels fulfilling. If you know you can just change a slider to grind less....

Unless you have settings like, 'dragons are my friends'[easy], 'money is not my object'[normal] and 'help capitalists took over my game'[hard] - then you feel like you've won your ship/fleet fair and square on a hard difficulty rather then "huh, I wonder if I have the patience to get my fleet on a 0.3x slower grind"

That said, I appreciate the feeling of wanting to slow the pace down by squeezing the money, especially if (but even if not) it's a well-calibrated (pseudo-)'realism' setting

**Comment by Zitchas** [comment]
True, those are good points. This probably leads to be something that would fit better tucked into gamerules that people have to actually seek out to change, rather than listed front-and-center to anyone to looks at the settings.

**Comment by desFritz** [comment]
> Increasing difficulty by increasing the grind is lazy and not good game design. As such, I 'd prefer it if difficulty was done by doing doing stuff like:

> This isn't really about making the game harder or easier. 

Agree. Difficulty was maybe the wrong way to write it, as it does not make the game more difficult. 
Usually you just find those setting under "diffuculty" when starting games.

> A problem I see with that is if a game forces you to grind, to a certain point, it feels fulfilling. If you know you can just change a slider to grind less....
> That said, I appreciate the feeling of wanting to slow the pace down by squeezing the money, especially if (but even if not) it's a well-calibrated (pseudo-)'realism' setting

That was more my point. 

- I want just to do combat, explore, board and do not care about money (200% income)
- I want the normal experience the game is balanced on (100%)
- I want to work (grind) for my money, fail the millionaire-in-30-days-challenge etc. ( < 100% income)

Would allow players to adjust it a bit to their pace (as an option), with just a bit of extra code. 
Just like the usual settings, you often find in the settings of games.

> This is also something that may fit better as an element of the gamerules system discussed by Amazinite here:
Haven't seen that one. 

Yes, income modifiers would make sense to go there.

**Comment by desFritz** [comment]
Sidenote:
I just put daily costs to hyperdrives, engines and power generator. Works too.

Default: I need outfit x?  Park ships, fly there, buy it. Travel does not cost anything,
with only your pilot. Haven't looked at the credit score for ages anyway.

Now: Oh, maintenance cost means parked ships cost 'parking fees', hm?
Maybe I can find it on a pirate ship, for that travel is 4x the outfit costs.
Oh, there is a job that gets me there? Accept.  

Default: I can just fly anywhere and buy stuff. Now I have to look at the credit score
and even take items from boarded ships that might be less valuable, but would cost
a fortune to purchase due to travel costs.
Just found a nitrogen cooler. Looted. With default income I would have just traveled
to an outfitter an bought it. Now my ships are more of a patchwork.

Maybe not the intention of the game, but that is what mods/plugins are for.

**Comment by Chopliver19** [comment]
Yes but don't the fuel and power come from renewable resources? Most power generators are nuclear, and most players can find a way around fuel using ramscoops. Unless you were talking about the upkeep of the power generator and hyperdrive. And of course, you could have better outfits that are renewable, while others aren't, and the price of the renewable ones could go up in price. With this thing that y'all are proposing, it could potentially make it a lot harder, but is optional if I heard it correctly? And @desFritz I have forgotten about my credit score a long time ago in Endless Sky.

**Comment by leklachu** [comment]
Next up: a mod where you can barely afford to fly a shuttle, because hyperdrive fuel is so expensive, but you're also sending a quarter of everything you earn back to your grandmother or she'll die of leukaemia, so you get married to have joint income, but your @!*%# husband drinks all your money away so you end up planet-bound and depressed until the war comes along and you get a job making billboard slogans for the Free Worlds and eventually divorce your husband and take to the skies again with a massive loan from the Free Worlds Investment Bank which you can never hope to pay off. The war ends but your bitterness and rage doesn't and you drag the Council and with it the Free Worlds into a vindictive war against the Syndicate and all "$!% men, eventually 'freeing' Syndicate space and replacing it with a communist regime with you in absolute power. Finally, skimming off 10% of the GDP of two thirds of the galaxy's planets, and having blown up the investment bank to which you would have owed your huge debt, you can now afford to upgrade to a Heavy Shuttle.

**Comment by desFritz** [comment]
True: Power generators are nuclear and such. But just because something is nuclear does not mean it does not require 'fuel', or even maintenance. I just hope our nuclear power plants are not run without regular maintenance. On the other hand: if they are floating through space... who cares ;-) It's a game anyway.

Speaking of game...

**Next up:** a mod where communism is real and you don't even need a credit score anymore. Could us get rid of such boring and useless things in this game like trading goods, doing missions and boarding/selling ships and focus on other things. Who writes the FR?

**Comment by leklachu** [comment]
> Next up: a mod where communism is real and you don't even need a credit score anymore.

"On your eighteenth birthday, the Glorious People's Republic gives you the median average of all galaxy spacecraft, which turns out to be a shuttle, and assigns you the role of transporting people every day. You will be paid in food and all fuel and maintenance costs will be provided for by the government. If you transgress your duties or fail to meet your quota you will be hunted down and put in a re-educational facility. Luckily, because you are a good citizen, you are happy with this life and continue to do it until you die."

More seriously, I quite like the idea of a difficulty mod that really balances out so you can fly a modest ship casually or a galaxy-destroying mega-fleet but with proportionate focus on earning enough money to keep it afloat. (Or a salary from the campaign.) Already in my first or second playthrough I enjoyed the scout then the bounder to explore around without feeling the instability of losing money every hyperspace jump.

I suppose it's already kind of that way, with crew salaries, except for my beautiful fleet of automated 349s, and the fact that piracy is so much more lucrative than 'honest' work. Perhaps a difficulty mod could focus on increasing crew salaries (to tighten that earning&killing-power/loss-power gap), further nerfing the market for second-hand outfits, and changing automation

- resale nerf: that might make for a really frustrating grind to high-expense ships, especially once in alien space, e.g. having big enough stuff for the Wanderers story. But maybe that's okay for the people who want this mod/setting
- The kor mereti/sestor lore-wise only function autonomously because of the collective consciousness and central control respectively. Maybe there could be some way to capture them and bring them back - but after that (or on boarding) you need to replace the control node/transciever with a manned "command center" (the one that allows you to fly as flagship) whereupon the crew requirement would go up to something proportionate to the complexity of the ship.
- then.... I like the idea of a mission chain that would have you set up your own command base whence to fly Kor Sestor ships autonomously... hm... :-)

A final couple of 'experience' thoughts, re: nerfing outfits

- I quite enjoyed my near-pacifist run, which meant amongst other things I couldn't plunder or capture ships. It forced me to focus on ordinary jobs more and take time for them, which I'd started to ignore otherwise
- That said, on my first game, when I tried out and discovered I could plunder pirate ships and make so much money off them, it was that awesome feeling of discovering a new huge thing - And Endless Sky is all about* those amazing discoveries every time you find a new thing.
  + *Well, and the various fun we manage to find after we think we've discovered everything in the game!

**Comment by MedysMKA** [comment]
Was not able to find this issue before so created a new one... Basically I second this idea.

Add a difficulty setting in form of a multiplier for how much money is paid for:
- Trading
- Jobs
- Missions
- Mining
- Tribute

**Comment by westhand** [comment]
Talking about the different ways to make money: mining does not make a lot of sense, lore wise.
1t of "metals" usually costs less than 500, 1t of "heavy metal" less than 1000. These are presumably bars. Who then pays 900 for 1t of lead ore?
These days, flotsam lives ages, which makes mining much less grindy than it used to be. So I would happily adjust its revenue to 10% or so, relative to everything else.
Also note, if normal economic activity becomes less lucrative, capturing wins out even more in comparison, unless depreciation is scaled accordingly.

**Comment by roadrunner56** [comment]
This has been somewhat resolved by #11302 in that there are now starts with different difficulty levels, but this is done through modifying the mortgage you start off with rather than changing how much money you make throughout the game. To be honest changing the payment and costs of everything in the game through a slider feels like it would run into balance issues very quickly, and isn't probably something we'd ever support.
